object/check.go

```diff
@@ -572,3 +578,55 @@ func CheckToEnableCaptcha(application *Application, organization, username strin
 
 	return false, nil
 }
+
+func CheckEntryIp(user *User, application *Application, organization *Organization, remoteAddress string, lang string) error {
+	entryIp, _, err := net.SplitHostPort(remoteAddress)
+	if err != nil {
+		return err
+	}
+
+	if user != nil && !isEntryIpAllowd(user.IpWhitelist, entryIp) {
+		return fmt.Errorf(i18n.Translate(lang, "check:Your IP address %s has been banned according to the configuration of user %s"), entryIp, user.Name)
+	}
+
+	if application != nil && !isEntryIpAllowd(application.IpWhitelist, entryIp) {
+		return fmt.Errorf(i18n.Translate(lang, "check:Your IP address %s has been banned according to the configuration of application %s"), entryIp, application.Name)
```

Comment:
`has been banned according to the configuration of: %s"`

So we don't need three i18n texts